verifier/token/cctp/attestation.go

```diff
@@ -2,42 +2,169 @@ package cctp
 	ctx context.Context,
 	txHash protocol.ByteSlice,
 	message protocol.Message,
-) (protocol.ByteSlice, error) {
-	// Call CCTPv2 using sourceChainDomain + txHash
-	// Find the attestation with the results set and return
-	// 		hookData can be found in attestation response - decodedMessage.decodedMessageBody.hookData
-	//      hookData format is the following <4 byte verifier version><32 byte msg ID>
-	return txHash, nil
+) (Attestation, error) {
+	sourceDomain, ok := sourceDomains[uint64(message.SourceChainSelector)]
+	if !ok {
+		return Attestation{}, fmt.Errorf("unsupported source chain selector: %d", message.SourceChainSelector)
+	}
+
+	response, err := h.client.GetMessages(ctx, message.SourceChainSelector, sourceDomain, txHash.String())
+	if err != nil {
+		return Attestation{}, fmt.Errorf(
+			"error fetching messages for chain selector %d and tx hash %s: %s",
+			message.SourceChainSelector, txHash, err,
+		)
+	}
+	return h.extractAttestationFromResponse(response, message)
+}
+
+func (h *HTTPAttestationService) extractAttestationFromResponse(response Messages, message protocol.Message) (Attestation, error) {
+	for _, msg := range response.Messages {
+		ok := h.matchesMessage(msg, message)
+		if !ok {
+			h.lggr.Debugw("skipping CCTP message as it doesn't match CCIP message", "message", msg)
+			continue
+		}
+		if msg.IsAttestationComplete() {
+			return NewAttestation(h.ccvVerifierVersion, msg)
+		}
+		return Attestation{}, fmt.Errorf("attestation is not ready")
+	}
+	return Attestation{}, fmt.Errorf("no matching message found in response")
+}
+
+func (h *HTTPAttestationService) matchesMessage(msg Message, message protocol.Message) bool {
+	messageID, err := message.MessageID()
+	if err != nil {
+		return false
+	}
+
+	lggr := logger.With(h.lggr, "cctpMessage", message, "messageID", messageID)
+
+	if !msg.IsV2() {
+		lggr.Debugw("CCTP Attestation: Skipping message due to unsupported CCTP version")
+		return false
+	}
+
+	ccvAddress, ok := h.ccvAddresses[message.SourceChainSelector]
+	if !ok {
+		lggr.Debugw("CCTP Attestation: No CCV address configured for source chain selector",
+			"sourceChainSelector", message.SourceChainSelector)
+		return false
+	}
+
+	senderAddress := protocol.UnknownAddress(msg.DecodedMessage.Sender)
+	if !ccvAddress.Equal(senderAddress) {
+		lggr.Debugw("CCTP Attestation: Skipping message due to sender address mismatch",
+			"expectedSender", ccvAddress.String(),
+			"actualSender", senderAddress.String(),
+		)
+		return false
+	}
+
+	actualHookData, err := protocol.NewByteSliceFromHex(msg.DecodedMessage.DecodedMessageBody.HookData)
+	if err != nil {
+		lggr.Debugw("CCTP Attestation: Skipping message due to invalid hook data",
+			"hookData", msg.DecodedMessage.DecodedMessageBody.HookData,
+			"error", err,
+		)
+		return false
+	}
+
+	// <4 byte verifier version><32 byte msg ID>
+	var expectedHookData protocol.ByteSlice
+	expectedHookData = append(expectedHookData, h.ccvVerifierVersion...)
```

Comment:
@kylesmartin Do we need to compare `ccvVerifierVersion` with the one returned from hookData? Right now, `ccvVerifierVersion` is hardcoded in the off-chain, but I guess at some stage we might have a different version? Or should we ignore comparing and just take whatever is returned from Iris as long as other fields match (messageId and sender)